modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/LogicalRelImplementor.java

```diff
@@ -172,6 +172,21 @@ public LogicalRelImplementor(
 
         Node<Row> input = visit(rel.getInput());
 
+        if (distribution.function().affinity()) { // Affinity key can't be null, so filter out null values.
+            assert distribution.getKeys().size() == 1 : distribution.getKeys().size();
+
+            int affKey = distribution.getKeys().get(0);
+
+            if (rel.getRowType().getFieldList().get(affKey).getType().isNullable()) {
```

Comment:
What if `get(affKey)` returns null?

Reply:
I think it's not possible, since `affKey` should be valid column index rowType contains all required columns.